### PR TITLE
[Feature] Initial multi-tests support in the Tester

### PR DESCRIPTION
**Pull request by vmarois**
In this PR, I am adding support for multi-tests  in `Tester`. It is basic, and somewhat constrained, but it worked for `SimpleConvNet/MNIST`. 

If the user would like to run multiple tests, he can use the `multi_tests` key in the `testing` section to indicate the keys which associated values will be different for each test config:

> problem:
    batch_size: 64
    data_folder: ~/data/mnist
    name: MNIST
    resize: [32, 32]
    use_train_data: false
  sampler: {}
  **multi_tests: {batch_size: [500, 1000]}**

The following constraints apply for now:

- Assume that the indicated varying values are **_leafs_** of the `testing` section
- The number of indicated varying values per key is the same for all keys
- The indicated order of the varying values will be respected, i.e. \

> multi_tests: {batch_size: [64, 128], img_size: [128, 256]}

and

> multi_tests: {batch_size: [64, 128], img_size: [256, 128]}

will lead to different test configs.

- At least one key has varying values (but this is implicit).

I am mainly opening this PR for code review & feedback, as this is still very basic for now. I want to test how this plays with more complex configs (e.g. when `sampler` is present).

### CODE REVIEW

**Comment by vmarois** [comment]
@tkornut I've pushed some commits that:
- Remove the `multi_test` key from the config so that the `testing_configuration.yaml` in each test folder does not contain it,
- Properly update the `logger.FileHandler` of the `Tester` so that `tester.log` in each test folder only contains the logs of the current experiment (wasn't the case previously),
- Pretty print some info about the current test experiment:

pretty_print_info

@aasseman : The bug where a `config` param is also present in the `default` ones is still there.. So only doing `params.del_config_param(key='config_key')` does not work on its own, we have to add `params.del_default_param(key='config_key')`. That's how I am getting around it, so let's not spend time on this bug for now (as this is the only place in `mip` where we need this feature). I'm opening an issue for reference though.

**Comment by vmarois** [comment]
Update: I'm fairly confident that I have fixed the bug where adding `config` params would also add them to the `default` ones. 
@tkornut Can you review that PR?

File: miprometheus/workers/tester.py [hunk 1]
```diff
@@ -356,9 +470,31 @@ def main():
     """
     tester = Tester()
     # parse args, load configuration and create all required objects.
-    tester.setup_experiment()
-    # GO!
-    tester.run_experiment()
+    tester.setup_global_experiment()
+
+    if tester.check_multi_tests():
```
Comment (tkornut) [inline miprometheus/workers/tester.py]: Ok, so this is how you solved it, i.e. outside of tester...

Comment (vmarois) [reply miprometheus/workers/tester.py]: Yes, had to check it outside of `setup_experiment` as I did not want to introduce too much overhead. In the case where the user does not indicate `multi_tests` (i.e. in most current config files), then the overhead is just the run time of `try -> except`, which should be negligible.

Comment (tkornut) [reply miprometheus/workers/tester.py]: sure, makes sense to me 👍

**Pull request merged**